glean-core/src/lib_unit_tests.rs

```diff
@@ -871,3 +871,38 @@ fn records_database_file_size() {
     // We should see the database containing some data.
     assert!(data.sum > 0);
 }
+
+#[test]
+fn records_io_errors() {
+    use std::fs;
+    let _ = env_logger::builder().is_test(true).try_init();
+
+    let (glean, _data_dir) = new_glean(None);
+    let pending_pings_dir = glean.get_data_path().join(crate::PENDING_PINGS_DIRECTORY);
```

Comment:
Not using `_data_dir`?

Also, the fact there's two `new_glean` test `fn` in the SDK is annoying. It was hard for me to find the right one.

Reply:
I was using `_data_dir` but then noticed there's a `get_data_path` function, so it seemed better to just use that.

About the duplication I agree, maybe it's time to clean up these tests.